CMakeLists.txt

```diff
@@ -163,6 +167,17 @@ if(BUILD_DEPENDENCIES)
       build_dependency(mbedtls ${BUILD_ARGS})
   endif()
 
+  # This step is necessary because the next set of dependencies have a dependency on
+  # mbedtls/openssl and the find_package command  populates certain cmake variables which
+  # are needed to pass to the cmake commands for the next set of dependencies
+  if (USE_OPENSSL)
```

Comment:
We should be able to remove the existing `find_package` calls on the crypto libs that start down on line 262 right?

Reply:
No because this is in the if build dependencies section, so this needs to run in both cases and since it's idempotent I kept it there.

Reply:
Sounds good, might want to consider breaking up the build_deps case to handle the before find_package logic and the after find_package logic.